### PR TITLE
Add an autoload spec for the behavior inside autoload but after the constant is defined

### PR DESCRIPTION
**Pull request by casperisfine**
From discussion on https://bugs.ruby-lang.org/issues/20188

cc @fxn @eregon

### CODE REVIEW

File: spec/ruby/core/module/autoload_spec.rb [hunk 1]
```diff
@@ -403,6 +403,66 @@ def check_before_during_thread_after(&check)
...
+      begin
+        ModuleSpecs::Autoload::DuringAutoload
+      ensure
+        in_loading_thread, in_other_thread = t.value
+      end
+      after = check.call
+      [before, in_loading_thread, in_other_thread, after]
+    end
+
+    it "returns nil in autoload thread and 'constant' otherwise for defined?" do
```
Comment (eregon) [inline spec/ruby/core/module/autoload_spec.rb:437]: This spec description needs to be adapted to reflect the results in this case

File: spec/ruby/core/module/autoload_spec.rb [hunk 2]
```diff
@@ -403,6 +403,66 @@ def check_before_during_thread_after(&check)
...
+    end
+
+    it "keeps the constant in Module#constants" do
+      results = check_before_during_thread_after {
+        ModuleSpecs::Autoload.constants(false).include?(:DuringAutoload)
+      }
+      results.should == [true, true, true, true]
+    end
+
+    it "returns false in autoload thread and true otherwise for Module#const_defined?" do
```
Comment (eregon) [inline spec/ruby/core/module/autoload_spec.rb:451]: This spec description needs to be adapted to reflect the results in this case

**Comment by eregon** [comment]
BTW it would be great to also test the case of the constant is set and then an exception happens in the autoload.

**Comment by casperisfine** [comment]
I fixed the naming issues.

I'll look into the error thing but no promises.

**Review by eregon — APPROVED** [review]

**Comment by fxn** [comment]
❤️ I'll have a stab at the spec for the error.

**Pull request merged**